corehq/apps/users/views/mobile/users.py

```diff
@@ -1210,25 +1211,44 @@ def page_url(self):
         return reverse(self.urlname, args=self.args, kwargs=self.kwargs)
 
 
-@method_decorator([FILTERED_BULK_USER_DOWNLOAD.required_decorator()], name='dispatch')
 class FilteredUserDownload(BaseManageCommCareUserView):
-    urlname = 'filter_and_download_commcare_users'
-    page_title = ugettext_noop('Filter and Download')
+    page_title = ugettext_noop('Filter and Download Users')
 
-    @method_decorator(require_can_edit_commcare_users)
     def get(self, request, domain, *args, **kwargs):
-        form = CommCareUserFilterForm(request.GET, domain=domain, couch_user=request.couch_user)
+        form = UserFilterForm(request.GET, domain=domain, couch_user=request.couch_user, user_type=self.user_type)
         # To avoid errors on first page load
         form.empty_permitted = True
         context = self.main_context
-        context.update({'form': form, 'count_users_url': reverse('count_users', args=[domain])})
+        context.update({'form': form, 'count_users_url': reverse(self.count_view, args=[domain])})
         return render(
             request,
             "users/filter_and_download.html",
             context
         )
 
 
+@method_decorator([require_can_use_filtered_user_download], name='dispatch')
+class FilteredCommCareUserDownload(FilteredUserDownload):
+    urlname = 'filter_and_download_commcare_users'
+    user_type = MOBILE_USER_TYPE
```

Comment:
I am still reviewing with few commits to go, and i saw the change from "include_" to user types, and i am still thinking that classes would be better. There is still a lot of "ifs" in the code which isn't easy to follow and is making things less cleaner.

Reply:
> I did switch from flags to types in this PR, but that was largely because my flags had been named include_* which was poor naming since it didn't make it clear the flags are mutually exclusive.

using "only_" would have solved that but i still think passing user types was better for other reasons.